Optimize bundling for overview and project-overview

TheRealJon opened the pull request:
This is mostly just clean up and organization from the original implementation of the overview side panel. The main motivation was to ensure that the `overview.jsx` and `project-overview.jsx` files do not get included in the main bundle.

- Rename `nav-title.tsx` to `headings.tsx` to make it more semantically accurate.
- Move the ResourceOverviewHeading component into `headings.tsx` (fixes bundling issue)
- Rename NavTitle component to `PageHeading` to make it more semantically accurate
- Delete `section-heading.tsx` file and move `SectionHeading` component into `headings.tsx`
- Rename all instances of NavTitle to PageHeading
- In touched files, break down long import statements into multiple lines, sort alphabetically (clean up)

alecmerdler commented on frontend/public/components/utils/headings.tsx:
```diff
@@ -61,19 +66,39 @@ export const NavTitle = connectToModel((props: NavTitleProps) => {
 
+export const SectionHeading: React.SFC<SectionHeadingProps> = ({text, children, style}) => <h2 className="co-section-heading" style={style}>{text}{children}</h2>;
+
+export const ResourceOverviewHeading = ({kindObj, actions, resource}) => <div className="co-m-nav-title resource-overview__heading">
```
`ResourceOverviewHeading: React.SFC<ResourceOverviewHeadingProps>`

TheRealJon commented:
Oops, thanks for catching that @alecmerdler!

TheRealJon commented:
@alecmerdler Addressed your comments. Yeah, I just don't like the long import lines because it makes it more difficult to see everything that's being imported, especially if there are unused or incorrect imports. I've just been fixing those as I come across them.

spadgett commented on frontend/public/components/utils/headings.tsx:
```diff
@@ -61,19 +67,45 @@ export const NavTitle = connectToModel((props: NavTitleProps) => {
+export type ResourceOverviewHeadingProps = {
+  kindObj: K8sKind,
+  actions: CogAction[],
+  resource: K8sResourceKind
```
These lines should end in `;` instead of `,`

TheRealJon commented:
@spadgett, fixed the semicolons in the the type declaration. Also, looked into those eslint rules for imports. Unfortunately, there are no rules to enforce new lines between each member of multi-member import statements. So, you can enforce this styling:

``` js
import {
  foo, bar
} from './baz';
```

but not this:
``` js
import {
  foo,
  bar
} from './baz';

Additionally, the `sort-imports` rule would not only sort members of a multi-member import, but also sort all imports relative to each other. Since we are currently grouping imports by external/internal source, this rule wouldn't work. There is a plugin that can be used to enforce grouping of imports: https://github.com/benmosher/eslint-plugin-import/blob/HEAD/docs/rules/order.md, but this would not really help with sorting.

spadgett commented:
@jhadvig FYI, looks like we might have a secret test flake.

https://s3.amazonaws.com/teamui-jenkins/jenkins-console-build-8824/GUI-Tests.log

```
1) Interacting with the create secret forms Webhook secrets : edits webhook secret
   NoSuchElementError: No element found using locator: by.buttonText("Generate")
```

The weird thing is the button is clearly visible in the captured screenshot for the failed test, so I'm not sure what's wrong.

spadgett commented:
jenkins rebuild
